.github/workflows/docker_publish.yml

```diff
@@ -47,7 +44,7 @@ jobs:
           stages: base_python, moab, dagmc
           server-stage: openmc
           quiet: false
-          tag-latest-on-default: false
+          tag-latest-on-default: true
```

Comment:
I think we want this to be set conditionally depending on whether it's a PR or a push

Reply:
This is what I'm thinking now:
`push`: we keep it how it used to be, with `tag-latest-on-default` as false, using the `DOCKER_IMAGE_TAG` env variable as well as the step that exports it to the `output` of the first job
`pull_request`: we set `tag-latest-on-default` to be true. But then, we would still need the step that exports `DOCKER_IMAGE_TAG` to `outputs:` because the tag for the image would change from `refs-heads...bk-0` to `latest`, and that change needs to be reflected in `container:` of the `BuildTest` job. Not to mention that we would need the if statement that would change `DOCKER_IMAGE_TAG` to `latest`, in addition to the if statement to change the value of `tag-latest-on-default`.
Overall, I'm not sure what setting `tag-latest-on-default` conditionally does that we didn't already have with the commit I did yesterday, and if anything, it seems more complex to me